src/shared/static/static.service.ts

```diff
@@ -1,14 +1,69 @@
 import { Injectable, Logger } from '@nestjs/common';
 import path from 'path';
-import { writeFileSync, readFileSync, unlink, mkdirSync, existsSync } from 'fs';
+import {
+  writeFileSync,
+  readFileSync,
+  unlink,
+  mkdirSync,
+  existsSync,
+  createWriteStream,
+  unlinkSync,
+  statSync,
+  readdir,
+} from 'fs';
 import { PNG, PNGWithMetadata } from 'pngjs';
 
+import { S3Client, PutObjectCommand, DeleteObjectCommand, GetObjectCommand } from '@aws-sdk/client-s3';
+import { Readable } from 'stream';
+
 export const IMAGE_PATH = 'imageUploads/';
 
 @Injectable()
 export class StaticService {
```

Comment:
we need to refactor this service to be a common entry point with generic methods 
could you make it the same way how it's done for `CompareService` 
this will make the code much cleaner and allow to extend it in future